### PR TITLE
Remove old object controller and the flag

### PR DESCRIPTION
**Pull request by yiqigao217**
The old object controller is removed. The flags to toggle the old/new
object controllers are also cleaned.

Tested by 'make test' and manual testing on GKE cluster.

### CODE REVIEW

File: incubator/hnc/config/rbac/role.yaml [hunk 1]
```diff
@@ -13,13 +13,10 @@ rules:
   verbs:
   - create
   - delete
-  - deletecollection
   - get
-  - impersonate
   - list
   - patch
   - update
-  - use
```
Comment (yiqigao217) [inline incubator/hnc/config/rbac/role.yaml]: Not sure if this needs to be checked in.

Comment (adrianludwin) [reply incubator/hnc/config/rbac/role.yaml]: Weird. I think I removed those permissions a while ago and just forgot to regenerate the files? Maybe remove this from the commit if you can and resubmit separately?

Comment (yiqigao217) [reply incubator/hnc/config/rbac/role.yaml]: Done.

**Pull request merged**